Kickstarter-iOS/Features/PledgePaymentMethods/Controller/NoShippingPledgeRewardsSummaryViewController.swift

```diff
@@ -0,0 +1,310 @@
+    view.frame = CGRectMake(0, 0, self.tableView.frame.size.width, self.tableView.frame.size.height)
+  }
+
+  private func sectionHeaderLabelStyle(_ label: UILabel) {
+    label.font = UIFont.ksr_subhead().bolded
+    label.textColor = UIColor.ksr_black
+    label.numberOfLines = 0
+    label.frame = CGRectMake(
+      CheckoutConstants.PledgeView.Inset.leftRight,
+      0,
+      self.tableView.frame.size.width,
+      PledgeRewardsSummaryStyles.Layout.sectionHeaderLabelHeight
+    )
+  }
+
+  private func tableViewContainerStyle(_ view: UIView) {
+    view.translatesAutoresizingMaskIntoConstraints = false
+    view.clipsToBounds = true
+  }
+
+  // MARK: - Helpers
+
+  private func setEntireViewToIsHidden(_ isHidden: Bool) {
+    self.view.isHidden = isHidden
+    self.pledgeTotalViewController.view.isHidden = isHidden
+    self.separatorView.isHidden = isHidden
+  }
+
+  // MARK: Apply DataSource Snapshot
+
+  private func applySnapshotToDataSource(from data: [PostCampaignRewardsSummaryItem]) {
+    var snapshot = NSDiffableDataSourceSnapshot<PledgeRewardsSummarySection, PledgeRewardsSummaryRow>()
+
+    /// Decipher header vs reward objects from the `[PostCampaignRewardsSummaryItem]` data object.
+    let headerItemData = data.compactMap { item -> PledgeExpandableHeaderRewardCellData? in
+      guard case let .header(data) = item else { return nil }
+      return data
+    }
+
+    let rewardData = data.compactMap { item -> PledgeExpandableHeaderRewardCellData? in
+      guard case let .reward(data) = item else { return nil }
+      return data
+    }
+
+    // MARK: Header Section
+
+    /// Define the sections of the table and what data to use in each section.
+    snapshot.appendSections([.header])
+    snapshot.appendItems([.header(headerItemData[0])], toSection: .header)
+
+    // MARK: Reward Section
+
+    let baseReward = rewardData[0]
+    snapshot.appendSections([.reward])
+    snapshot.appendItems([.reward(baseReward)], toSection: .reward)
+
+    // MARK: Add-Ons
+
+    let addOns = rewardData
+      .filter { reward in reward != baseReward && reward.text != Strings.Bonus_support() }
+      .map { PledgeRewardsSummaryRow.addOns($0) }
+
+    /// We only want to add an add-ons section if any have been selected
+    if addOns.count > 0 {
+      snapshot.appendSections([.addOns])
+      snapshot.appendItems(addOns, toSection: .addOns)
+    }
+
+    // MARK: Bonus Support
+
+    let bonusSupportReward = rewardData
+      .filter { reward in reward.text == Strings.Bonus_support() }
+      .map { PledgeRewardsSummaryRow.bonusSupport($0) }
+
+    if let bonusSupport = bonusSupportReward.first {
+      snapshot.appendSections([.bonusSupport])
+      snapshot.appendItems([bonusSupport], toSection: .bonusSupport)
+    }
+
+    /// Applys the changes above as a snapshot of all of the data needed to render the `UITableView `.
+    self.dataSource.apply(snapshot, animatingDifferences: true)
```

Comment:
This method defines the sections and data used in the table via snapshot.appendSections and snapshot.appendItems, respectively. Then, it updates the table all at once using apply.